frontend/src/views/Org/MembersPage/components/OrgMachineIdentityTab/components/MachineIdentitySection/CreateClientSecretModal.tsx

```diff
@@ -0,0 +1,283 @@
+                                            isLoading={isSubmitting}
+                                            isDisabled={isSubmitting}
+                                        >
+                                            Create
+                                        </Button>
+                                    </div>
+                                </FormControl>
+                            )}
+                        />
+                    </form>
+                )}
+                <h2 className="mb-4">Client Secrets</h2>
+                <TableContainer>
+                    <Table>
+                        <THead>
+                            <Tr>
+                                <Th>Description</Th>
+                                <Th>Expires At</Th>
+                                <Th>Client Secret</Th>
+                                <Th className="w-5" />
+                            </Tr>
+                        </THead>
+                        <TBody>
+                            {isLoading && <TableSkeleton columns={4} innerKey="org-machine-identities-client-secrets" />}
+                            {!isLoading &&
+                            data &&
+                            data.length > 0 && 
+                            data.map(({
+                                _id,
+                                description,
+                                machineIdentity,
+                                expiresAt,
+                                clientSecretPrefix
+                            }) => {
+                                return (
+                                    <Tr className="h-10" key={`mi-client-secret-${_id}`}>
+                                        <Td>{description === "" ? "-" : description}</Td>
+                                        <Td>{expiresAt ? format(new Date(expiresAt), "yyyy-MM-dd") : "-"}</Td>
+                                        <Td>{`${clientSecretPrefix}************`}</Td>
+                                        <Td className="flex">
+                                            <IconButton
+                                                onClick={async () => {
+                                                    await deleteClientSecretMutateAsync({
+                                                        machineId: machineIdentity,
+                                                        clientSecretId: _id
+                                                    });
+                                                }}
+                                                size="lg"
+                                                colorSchema="primary"
+                                                variant="plain"
+                                                ariaLabel="update"
+                                                className="ml-4"
+                                            >
+                                                <FontAwesomeIcon icon={faXmark} />
```

Comment:
a pop up to confirm delete would be good here

Reply:
I wasn't sure about the pop-up on top of pop-up UX pattern so I omitted it for now; I can add it.